---
title: C6310 | Microsoft Docs
ms.custom: ''
ms.date: 11/04/2016
ms.technology:
- vs-ide-code-analysis
ms.topic: conceptual
f1_keywords:
- C6310
helpviewer_keywords:
- C6310
ms.assetid: e759eb63-883f-4c3e-bf2f-b924ff511405
author: mikeblome
ms.author: mblome
manager: douge
ms.workload:
- multiple
ms.openlocfilehash: 59be8a75e0f68ce1147596606cf56e7af53a37e7
ms.sourcegitcommit: 6a9d5bd75e50947659fd6c837111a6a547884e2a
ms.translationtype: MT
ms.contentlocale: de-DE
ms.lasthandoff: 04/16/2018
---
# <a name="c6310"></a>C6310
Warnung C6310: Eine unzulässige Konstante im Ausnahmefilter kann unerwartetes Verhalten verursachen.  
  
 Diese Meldung gibt an, dass eine ungültige Konstante in der Filter-Ausdruck von einem strukturierten Ausnahmehandler erkannt wurde. Die Konstanten für die Verwendung in einem strukturierten Ausnahmehandler Filterausdruck definiert werden:  
  
-   `EXCEPTION_CONTINUE_EXECUTION`  
  
-   `EXCEPTION_CONTINUE_SEARCH`  
  
-   `EXCEPTION_EXECUTE_HANDLER`  
  
 Diese Werte werden in der Common Language Runtime-Header-Datei excpt.h definiert.  
  
 Eine Konstante, die nicht in der vorangehenden Liste aufgeführt ist, kann zu unerwartetem Verhalten führen.  
  
## <a name="example"></a>Beispiel  
 Der folgende Code generiert diese Warnung:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (EXCEPTION_ACCESS_VIOLATION)  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 So korrigieren Sie die Warnung unter Verwendung des folgenden Codes  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)  
  {  
    puts("Exception Occurred");  
  }  
}  
```